### ISSUE TITLE
Unable to run `Pkg.add("Lint")`

### ISSUE THREAD

**Issue by ricardoaiglesias** [tonyhffong/Lint.jl#254]
Good afternoon. I'm trying to install the linter package by running:

```
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.0.1 (2018-09-29)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using Pkg
```

But when I run the following command, I get an `Unsatisfiable requirements` error (complete stacktrace attached)

`julia> Pkg.add("Lint")`
[Stacktrace.txt](https://github.com/tonyhffong/Lint.jl/files/2440382/Stacktrace.txt)

I'm following the [installation guide here](https://github.com/tonyhffong/Lint.jl), and am running Julia 1.0 on a Mac OS X, Version 10.13.6 if that helps at all. 
Thanks in advance!

**Comment by ikselven** [comment]
The `Unsatisfiable requirements` message is due to [changes in the general package registry](https://discourse.julialang.org/t/package-compatibility-caps/15301) to prevent Pkg in Julia 1.0 from installing packages that are incompatible with Julia 1.0

In other words: Lint.jl does not support Julia 1.0, as the comments in #252 indicate.

**Comment by gabrielpreviato** [comment]
If I get it right, the package was fixed to be 1.0 compatible, but it was not released yet, since the way packages are released now has been changed, and the people that are maintaining this package don't know how to do it. Unfortunately i don't know either. You can add an unregistered package, I have not tried this, but it may work. Look for the section "Adding unregistered packages" https://docs.julialang.org/en/v1.0/stdlib/Pkg/

**Comment by m-wells** [comment]
If you fork this package you can install it
```julia
pkg> add https://github.com/m-wells/Lint.jl
```
I encountered this same issue with [SkyCoords.jl](https://github.com/JuliaAstro/SkyCoords.jl/issues/22) and I discuss the fix there.  The [`Compat.toml`](https://github.com/JuliaRegistries/General/blob/master/L/Lint/Compat.toml) file on the [JuliaRegistries](https://github.com/JuliaRegistries/General) needs to be updated.  A trivial push to `Lint.jl` should trigger a `METADATA` sync and update the Registry.

**Comment by tobanw** [comment]
@m-wells I don't think a simple push will do it here because attobot isn't installed. It would be easiest if attobot could be enabled. The alternative is to use `PkgDev` but [it isn't 1.0 compatible yet](https://github.com/JuliaLang/PkgDev.jl/pull/144), though maybe you could just run it from 0.6? As for manually creating a pull request, the README for the [METADATA.jl repo](https://github.com/JuliaLang/METADATA.jl) states: "METADATA.jl should not be edited directly unless absolutely necessary in an emergency."

In any case, a release needs to be created. If attobot is connected, then it will automatically update the registry.

@TotalVerb is there a reason we can't just add attobot?

**Comment by TotalVerb** [comment]
I'd love to add attobot, but I'm not an admin in this repo. We can maybe fork it to a standard "Julia$Org" for some value of Org, and install attobot there. Any suggestions?

**Comment by tobanw** [comment]
Looks like @tonyhffong hasn't been active here for a while. It would be ideal to fork to an Org. Looking at the [list of github orgs](https://julialang.org/community/), it seems that [Julia Editor Support](https://github.com/JuliaEditorSupport) is the closest match -- but not a great match. It really belongs with other code analysis/testing tools (debugger, unit testing, etc). Maybe someone should create an org for that...

Also, if we fork, how can we transfer the `Lint.jl` name in the official registry from this repo to the new one?

In the meantime, do you want to try running `PkgDev` from 0.6 to submit to the registry? Or do you need admin privileges to cut a release?

**Comment by TotalVerb** [comment]
I believe you must make a METADATA.jl PR by hand to change the name, though I have not ever done this. I think we should do this instead of trying to run `PkgDev`; I don't have an  installation of 0.6 handy to do this manually.

**Comment by tobanw** [comment]
You can just [download the 0.6 binaries](https://julialang.org/downloads/oldreleases.html). Then extract and run the executable in the `bin` dir. It's completely standalone (doesn't require any system installation) and won't affect your current install.

**Comment by galenlynch** [comment]
I've done PRs to metadata by hand before and it's not that difficult. That being said, the easiest thing would probably be to just download the 0.6 binaries and use PkgDev to tag a new version.

Here's my recollection for how to tag a new version by hand:
1) create a new annotated git tag in the Lint repository with the version number you want to release, and push it to Lint.jl on github
2) Go to ~/.julia/v0.6/METADATA/ and pull the most recent version from the official repository
2) in ~/.julia/v0.6/METADATA/Lint/versions add a new folder with the release number (no v at the beginning)
3) in that release folder, copy the REQUIRE file from the tagged version of Lint to a filed named "require"
4) make a new file in the release folder, called "sha1", with the sha1 for the commit that is the release
5) add and commit the changes
6) push the changes to your own fork of METADATA, and then open a PR

If you want to change the URL of the repository, i.e. you want to host it a new org, change the url of ~/.julia/v0.6/METADATA/Lint/url. It would probably be best to make it a separate PR, and explain the reasoning for changing ownership of the repository.

**Comment by tobanw** [comment]
@TotalVerb do you want to try the `PkgDev` method using 0.6 binaries?

Also, with the [new debugger](https://julialang.org/blog/2019/03/debuggers), I noticed there's a new org: [JuliaDebug](https://github.com/JuliaDebug). Not quite a perfect fit, since it's just repos for the debugger, but could be a potential new home.

**Comment by TotalVerb** [comment]
Pinging @vchuravy, do you think JuliaDebug could host this? Or maybe do you feel like somewhere else is more appropriate?

**Comment by FelipeLema** [comment]
@tobanw I'll try the `PkgDev` + 0.6bin method. Will report in this thread.

**Comment by FelipeLema** [comment]
@TotalVerb I'll need a new tag pushed to this repo to continue. Can't push it myself.

May I suggest `v0.6.0` ?

**Comment by TotalVerb** [comment]
@FelipeLema done just now.

**Comment by FelipeLema** [comment]
I took the liberty of making explicit that we support julia 1.0 onwards (not 0.6) in https://github.com/JuliaLang/METADATA.jl/pull/22873/commits/1a036ae7cfa1bf8eb265c0280c4a72ff97f03225

I'd say that supporting previous versions would be a separate feature.

**Comment by jayvdb** [comment]
I dont see 0.6.0 in https://github.com/JuliaRegistries/General/blob/master/L/Lint/Versions.toml

**Comment by FelipeLema** [comment]
we're waiting on https://github.com/JuliaLang/METADATA.jl/pull/22873 to be accepted.

Feel free to comment there or to give it a thumbs up for maintainers to pick it up.

**Comment by FelipeLema** [comment]
I was just [pointed out](https://github.com/JuliaLang/METADATA.jl/pull/22873#issuecomment-498323611) that we're supposed to use [registration bot](https://github.com/JuliaRegistries/Registrator.jl#transitioning-from-require-to-projecttoml)

**Comment by Octogonapus** [comment]
Any update on this? I am on Julia 1.1 and get the same error.

**Comment by jayvdb** [comment]
While not really a solution, my workaround is to have a custom project and do

`install.julia.jl`:
```julia
if VERSION < v"0.7.0-DEV.5183"
  Pkg.clone(pwd(), ENV["JL_PKG"])
  Pkg.build(ENV["JL_PKG"])
else
  using Pkg
  Pkg.build()
end
```

`install.julia.sh`
```bash
julia -e 'using Pkg; Pkg.add(PackageSpec(url="https://github.com/tonyhffong/Lint.jl", rev="v0.6.0"))'
export JL_PKG=FooBar
julia --color=yes install.julia.jl
julia -e 'import Lint.lintfile'
```

**Comment by CalebUsadi** [comment]
I'm still having a little bit of a problem. When I run
```shell
pkg> add https://github.com/tonyhffong/Lint.jl/
```
`Lint v0.5.2+ #master` is installed. This appears not to be version 0.6.0, and I can't get it to work with linter-julia. Is this an experience other people are having? Or am I just misinterpreting 0.5.2+?

**Comment by logankilpatrick** [comment]
@CalebUsadi Try this: 
```
(v1.0) pkg> add Lint#v0.6.0
```

**Comment by CalebUsadi** [comment]
@logankilpatrick 

No changes! Is there a way to verify that perhaps the package itself _is_ 0.6.0?

**Comment by logankilpatrick** [comment]
I tried and that should work! You may need to do `rn Lint ` in the package manager.

**Comment by logankilpatrick** [comment]
And yes you can verify by looking at the repo and checking the most recent tagged release, which is 0.6.0.

**Comment by CalebUsadi** [comment]
For some reason I'm still ending up with Lint v0.5.2+. Not sure why, or how to resolve it.

**Comment by logankilpatrick** [comment]
Yes sorry.

**Comment by logankilpatrick** [comment]
If doing rm and then add Lint#0.6.0

**Comment by logankilpatrick** [comment]
Doesn’t fix it, then maybe the version is being required by one of the other packages you are using.

**Comment by CalebUsadi** [comment]
I removed linter-julia from Atom, unfortunately now when I get this:
```julia
(v1.1) pkg> add Lint#0.6.0
  Updating git-repo `https://github.com/tonyhffong/Lint.jl.git`
  Updating git-repo `https://github.com/tonyhffong/Lint.jl.git`
ERROR: git object 0.6.0 could not be found
(v1.1) pkg> add Lint
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package Lint [d4b2101a]:
 Lint [d4b2101a] log:
 ├─possible versions are: [0.0.1-0.0.22, 0.1.0-0.1.71, 0.2.0-0.2.5, 0.3.0, 0.4.0, 0.5.0-0.5.2] or uninstalled
 ├─restricted to versions * by an explicit requirement, leaving only versions [0.0.1-0.0.22, 0.1.0-0.1.71, 0.2.0-0.2.5, 0.3.0, 0.4.0, 0.5.0-0.5.2]
 └─restricted by julia compatibility requirements to versions: uninstalled — no versions left
```
I can add it back from the url with `add https://github.com/tonyhffong/Lint.jl`, but I end up with 0.6.0.

**Comment by logankilpatrick** [comment]
What’s your Julia version? 

Best, 
Logan Kilpatrick

**Comment by CalebUsadi** [comment]
I'm using 1.1.0, oh no! Am I out of luck unless I download an older version of Julia?

**Comment by logankilpatrick** [comment]
It works perfectly for me on v1.0.3... at least in regards to installing. I actually was not able to get any use out of the package itself, but I can at least install on Julia 1.0.3. 

Best, 
Logan Kilpatrick

**Comment by CalebUsadi** [comment]
Oh well, thank you!

**Comment by logankilpatrick** [comment]
Just installed a clean build of Julia on my computer and doing `] add Lint#v0.6.0` on Julia v1.1.1.1 worked! 

Best, 
Logan Kilpatrick

**Comment by CalebUsadi** [comment]
I did the same, now on 1.1.1 and still unable to find 0.6.0. Not sure what's preventing me, or what I can do to troubleshoot.

**Comment by FelipeLema** [comment]
now that #263 has been merged, we can update the package following the [registratorbot instructions](https://github.com/JuliaRegistries/Registrator.jl#how-to-use)

I believe @tonyhffong is the only one who can do this update, though

**Comment by CalebUsadi** [comment]
It seems like @tonyhffong hasn't been active here for a while. As was discussed earlier, should this repo be forked somewhere else?

**Comment by logankilpatrick** [comment]
I’m thinking of making my own Linter at this point tbh. 

Best, 
Logan Kilpatrick

**Comment by CalebUsadi** [comment]
Although I know next to nothing about how one would implement that, definitely update here if you do begin to go about doing that!

**Comment by logankilpatrick** [comment]
Im sure much can be learned from tis package.  I think there’s a need since the package does not work for me via the Atom Package or the pure Julia version in the command line. I’ll keep this thread posted as I build it out slowly over time. I will probably build out the feature set over time.

**Comment by TotalVerb** [comment]
Yes, the repo should be forked somewhere else or we should start anew. This linter is really out of date and has a lot of work required to actually make sense for Julia 1.0.

**Comment by logankilpatrick** [comment]
Never written a linter before, but I’m going to give it a shot! Check out www.github.com/logankilpatrick/linting.jl

Best, 
Logan Kilpatrick

**Comment by tobanw** [comment]
You may want to look into the [Language Server Protocol](https://langserver.org/), which can provide linting in addition to completions and other nice features. There's already a Julia implementation: [LanguageServer.jl](https://github.com/julia-vscode/LanguageServer.jl)

**Comment by logankilpatrick** [comment]
Wonderful. I’ll take a look. Writing a linter is hard so it will be nice to have this.

**Comment by aminya** [comment]
Someone just needs to register for this package. Although the behavior isn't perfect, the master branch works.

**Comment by Dario-Rosa85** [comment]
I am getting a sligthly different error

ERROR: expected package `Lint [d4b2101a]` to be registered

anyone knows the solution?

**Comment by FelipeLema** [comment]
@Dario-Rosa85 I believe you're better off trying out some of the alternatives to this package listed above in this issue-conversation

**Comment by Dario-Rosa85** [comment]
@FelipeLema  Thank you. I will
